omod/src/main/web/dashboardwidgets/programstatus/programstatus.controller.js

```diff
@@ -466,8 +463,42 @@ export default class ProgramStatusController {
             this.patientProgram.states = this.$filter('filter')(this.patientProgram.states, (state) => {
                 return !state.voided
             }, true);
-            this.patientProgram.states = this.$filter('orderBy')(this.patientProgram.states, (state) => {
-                return new Date(state.startDate);
+            this.patientProgram.states = this.$filter('orderBy')(this.patientProgram.states, null, false, function(state1, state2)
+             {
+            	var result = 0;
+            	
+            	//console.log(state1, state2);
+            	
+            	//if the start date of 1 is earlier, is occurs before 
+            	if(state1.value.startDate < state2.value.startDate) {
+            		//console.log("based on startDate", state1.value.startDate, "is before", state2.value.startDate);
+            		result = -1;
+            	//later it occurs after
+            	} else if (state1.value.startDate > state2.value.startDate) {
+            		result = 1;
```

Comment:
and a return 1 here... just let anyone following the code realize they don't need to read through further for this path.